swift/Workflow/Sources/SubtreeManager.swift

```diff
@@ -9,29 +9,28 @@ extension WorkflowNode {
 
         internal var onUpdate: ((Output) -> Void)? = nil
 
-        /// Signals from the outside world (e.g. UI)
-        private let (childEvent, childEventObserver) = Signal<Signal<Output, NoError>, NoError>.pipe()
+        /// Sinks from the outside world (i.e. UI)
+        private var sinks: [SinkContext] = []
 
         /// The current array of children
         private (set) internal var childWorkflows: [ChildKey:AnyChildWorkflow] = [:]
 
+        /// The current array of workers
         private (set) internal var childWorkers: [AnyChildWorker] = []
 
-        private let (lifetime, token) = Lifetime.make()
+        /// Subscriptions from the outside world.
+        private var subscriptionContext: SubscriptionContext = SubscriptionContext(eventSources: [])
 
-        init() {
-            childEvent
```

Comment:
This logic is moved into the `SubscriptionContext`. After the sink changes, it was the only thing still using this (since `Workers` and `Workflows` call the handler function). Mostly refactored to keep it consistent.